doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/defaultProcessors/IncludeArgDocProcessor.kt

```diff
@@ -0,0 +1,242 @@
+import nl.jolanrensen.docProcessor.javaLinkRegex
+import nl.jolanrensen.docProcessor.removeEscapeCharacters
+
+/**
+ * @see IncludeArgDocProcessor
+ */
+const val INCLUDE_ARG_DOC_PROCESSOR = "nl.jolanrensen.docProcessor.defaultProcessors.IncludeArgDocProcessor"
+
+/**
+ * Adds two tags to your arsenal:
+ * - `@arg argument content` to declare an argument with content that can be used in the same doc.
+ * - `@includeArg argument` to include an argument from the same doc.
+ *
+ * This can be useful to repeat the same content in multiple places in the same doc, but
+ * more importantly, it can be used in conjunction with [IncludeDocProcessor].
+ *
+ * For example:
+ *
+ * File A:
+ * ```kotlin
+ * /** NOTE: The {@includeArg operation} operation is part of the public API. */
+ *  internal interface ApiNote
+ * ```
+ *
+ * File B:
+ *```kotlin
+ * /**
+ *  * Some docs
+ *  * @include [ApiNote]
+ *  * @arg operation update
+ *  */
+ * fun update() {}
+ * ```
+ *
+ * File B after processing:
+ * ```kotlin
+ * /**
+ *  * Some docs
+ *  * NOTE: The update operation is part of the public API.
+ *  */
+ * fun update() {}
+ * ```
+ *
+ * NOTE: If there are multiple `@arg` tags with the same name, the last one processed will be used.
+ * The order is: Inline tags: depth-first, left-to-right. Block tags: top-to-bottom.
+ * NOTE: Use `[References]` as keys is you want extra refactoring-safety.
+ * They are queried and saved by their fully qualified name.
+ */
+class IncludeArgDocProcessor : TagDocProcessor() {
+
+    private val useArgumentTag = "includeArg"
+
+    private val declareArgumentTag = "arg"
+    override fun tagIsSupported(tag: String): Boolean =
+        tag in listOf(useArgumentTag, declareArgumentTag)
+
+    override fun shouldContinue(
+        i: Int,
+        anyModifications: Boolean,
+        parameters: ProcessDocsAction.Parameters,
+        filteredDocumentables: Map<String, List<DocumentableWrapper>>,
+        allDocumentables: Map<String, List<DocumentableWrapper>>
+    ): Boolean {
+        if (i >= parameters.processLimit)
+            onProcesError(filteredDocumentables, allDocumentables)
+
+        // We can break out of the recursion if there are no more changes. We don't need to throw an error if an
+        // argument is not found, as it might be defined in a different file.
+        if (i > 0 && !anyModifications) {
+            val argsNotFound = argsNotFound.flatMap { (documentable, args) ->
+                args.map { arg -> "`${documentable.path}` -> @$useArgumentTag $arg" }
+            }.joinToString("\n")
+            println("IncludeArgDocProcessor WARNING: Could not find all arguments:[\n$argsNotFound\n]")
+            return false
+        }
+
+        return super.shouldContinue(i, anyModifications, parameters, filteredDocumentables, allDocumentables)
+    }
+
+    // @arg map for path -> arg name -> value
+    private val argMap: MutableMap<DocumentableWrapper, MutableMap<String, String>> = mutableMapOf()
```

Comment:
No. `process` is called once per tag. `argMap` is used to store `@arg` values per documentable per key, so that when `@includeArg` is processed, the value from the map is taken.